hoomd/filter/tags.py

```diff
@@ -0,0 +1,26 @@
+from hoomd.filter.filter_ import ParticleFilter
+from hoomd._hoomd import ParticleFilterTags
+import numpy as np
+
+
+class Tags(ParticleFilter, ParticleFilterTags):
+    def __init__(self, tags):
+        if isinstance(tags, np.ndarray):
+            tags = tags.astype(np.uint32)
+        else:
+            tags = np.array(tags, dtype=np.uint32)
+        tags = np.unique(tags)
+        self._tags = tags
```

Comment:
One line to rule them all. `np.unique` will be sorted and unique, `ascontiguousarray` with `dtype` will enforce all the other constraints.
```suggestion
        self._tags = np.ascontiguousarray(np.unique(tags), dtype=np.uint32)
```

Reply:
Nice one-liner, and actually more clear than the 6 lines it replaces.

Reply:
However I will need to ensure that `ParticleFilterTags` is called with this array not the tags variable which has not been processed.